sklearn/compose/_column_transformer.py

```diff
@@ -303,11 +306,18 @@ def _validate_remainder(self, X):
                 "'passthrough', or estimator. '%s' was passed instead" %
                 self.remainder)
 
-        n_columns = X.shape[1]
+        # Make it possible to check for reordered named columns on transform
+        if hasattr(X, 'columns') \
+           and any(_check_key_type(cols, str) for cols in self._columns):
```

Comment:
```suggestion
               any(_check_key_type(cols, str) for cols in self._columns)):
```

Reply:
Thank you. Answering only here for all suggested changes: Applied all of them in one commit. For cases like here where the change causes the conditionals' indentation to be equal to that of the following code, I used your suggestion as a guide line and used extra indentation.